operator/internal/component/podclique/pod/task.go

```
@@ -0,0 +1,103 @@
+
+	"github.com/go-logr/logr"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// podCreationTask creates a utils.Task which will create a Pod, capture the create-expectation and also emit a success/failed event post creation.
+func (r _resource) podCreationTask(logger logr.Logger, pclq *grovecorev1alpha1.PodClique, podGangName, pclqExpectationsKey string, taskIndex, podHostNameIndex int) utils.Task {
+	pclqObjKey := client.ObjectKeyFromObject(pclq)
+	return utils.Task{
+		Name: fmt.Sprintf("CreatePod-%s-%d", pclq.Name, taskIndex),
+		Fn: func(ctx context.Context) error {
+			pod := &corev1.Pod{}
+			// build the Pod resource
+			if err := r.buildResource(pclq, podGangName, pod, podHostNameIndex); err != nil {
+				return groveerr.WrapError(err,
+					errCodeBuildPodResource,
+					component.OperationSync,
+					fmt.Sprintf("failed to build Pod resource for PodClique %v", pclqObjKey),
+				)
+			}
+			// create the Pod
+			if err := r.client.Create(ctx, pod); err != nil {
+				if apierrors.IsAlreadyExists(err) {
+					logger.Info("pod creation failed as it already exists, ignoring the error", "pod", pod.Name)
+					return nil
+				}
+				r.eventRecorder.Eventf(pclq, corev1.EventTypeWarning, groveevents.ReasonPodCreationFailed, "Error creating pod %v: %v", pod.Name, err)
+				return groveerr.WrapError(err,
+					errCodeCreatePod,
+					component.OperationSync,
+					fmt.Sprintf("failed to create Pod: %s for PodClique %v", pod.Name, pclqObjKey),
+				)
+			}
+			logger.Info("Created Pod for PodClique", "podName", pod.Name, "podUID", pod.GetUID())
+			if err := r.expectationsStore.ExpectCreations(logger, pclqExpectationsKey, pod.GetUID()); err != nil {
+				utilruntime.HandleErrorWithLogger(logger, err, "could not record create expectations for Pod", "pclqObjKey", pclqObjKey, "pod", pod.Name)
```

what would be consequence if we can't record this expectation ?
Here we are just logging the error, shouldn't we return an error ?

---

To add:
I am at least logging the error.
Look at the k8s code base - they are completely ignoring it:
https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/replicaset/replica_set.go#L614 :)